Yearn-agents/Protocol-Specific-Agents/src/agent.ts

```diff
@@ -0,0 +1,214 @@
+  BlockEvent,
+} from 'forta-agent';
+import {
+  decodeParameters,
+  provideEventCheckerHandler,
+  provideFunctionCallsDetectorHandler,
+} from 'forta-agent-tools';
+import Web3 from 'web3';
+import MakerFetcher from './maker.fetcher';
+import TimeTracker from './time.tracker';
+import {
+  checkOSMContracts,
+  createFindingStabilityFee,
+  createOSMPriceFinding,
+  createStaleSpotFinding,
+  decodeNumber,
+  getCollateralType,
+  JUG_CONTRACT,
+  JUG_DRIP_FUNCTION_SIGNATURE,
+  OSM_CONTRACTS,
+  POKE_SIGNATURE,
+  SPOT_ADDRESS,
+} from './utils';
+
+const web3: Web3 = new Web3(getJsonRpcUrl());
+const fetcher = new MakerFetcher(web3);
+
+// The Handler that checks Stability Fee changes for Maker Strategies
+const provideStabilityFeeHandler = async (
+  web3: Web3,
+  makers: string[],
+  txEvent: TransactionEvent
+): Promise<Finding[]> => {
+  const findings: Finding[] = [];
+  const promises: any = [];
+
+  if (!txEvent.status) return [];
+
+  makers.forEach((strategy) => {
+    promises.push(
+      getCollateralType(web3, strategy, txEvent.blockNumber).then((res) => {
+        return {
+          strategy: strategy,
+          collateralType: res,
+        };
+      })
+    );
+  });
+
+  const collaterals: any = (await Promise.all(promises)).flat();
+
+  for (const res of collaterals) {
+    const filterOnArguments = (args: { [key: string]: any }): boolean => {
+      return args[0] === res.collateralType;
+    };
+
+    const agentHandler = provideFunctionCallsDetectorHandler(
+      createFindingStabilityFee(res.strategy),
+      JUG_DRIP_FUNCTION_SIGNATURE,
+      { to: JUG_CONTRACT, filterOnArguments }
+    );
+
+    findings.push(...(await agentHandler(txEvent)));
+  }
+
+  return findings;
+};
+
+// The Handler that checks poke() method is not called for >= 3 hours
+const provideStaleSpotPriceHandler = async (
+  web3: Web3,
+  makers: string[],
+  txEvent: TransactionEvent,
+  timeTracker: TimeTracker
+): Promise<Finding[]> => {
+  const findings: Finding[] = [];
+  const promises: any = [];
+  const timestamp = txEvent.block.timestamp;
+
+  makers.forEach((strategy) => {
+    promises.push(
+      getCollateralType(web3, strategy, txEvent.blockNumber).then((res) => {
+        return {
+          strategy: strategy,
+          ilk: res,
+        };
+      })
+    );
+  });
+
+  const collaterals: any = (await Promise.all(promises)).flat();
+
+  for (const res of collaterals) {
+    const filterEvents = (log: Log) => {
+      const parameters = decodeParameters(
+        ['bytes32', 'bytes32', 'uint256'],
+        log.data
+      );
+
+      return res.ilk === parameters[0];
+    };
+
+    const handler = provideEventCheckerHandler(
+      () => {
+        return {} as Finding;
+      },
+      POKE_SIGNATURE,
+      SPOT_ADDRESS,
+      filterEvents
+    );
+
+    if (!timeTracker.isIn3Hours(res.strategy, timestamp)) {
+      timeTracker.updateFindingReport(false);
+      timeTracker.updateFunctionWasCalled(false);
+    }
```

Comment:
Do you need this?

Reply:
Again, it's pointless to having them too :(